src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -80,6 +93,128 @@ public void parseName_validValueWithWhitespace_returnsTrimmedName() throws Excep
+                VALID_MCQ_BACK, (Set<String>) null));
+    }
+
+    @Test
+    public void parseMcqCard_invalidValue_throwsIllegalValueException() {
+        Assert.assertThrows(IllegalValueException.class, () -> ParserUtil.parseMcqCard(VALID_MCQ_FRONT,
+                INVALID_MCQ_BACK_INT, VALID_MCQ_OPTIONS));
+        Assert.assertThrows(IllegalValueException.class, () -> ParserUtil.parseMcqCard(VALID_MCQ_FRONT,
+                INVALID_MCQ_BACK_VALUE, VALID_MCQ_OPTIONS));
+    }
+
+    @Test
+    public void parseMcqCard_validValue_returnsMcqCard() throws Exception {
+        McqCard expectedMcqCard = new McqCard(VALID_MCQ_FRONT, VALID_MCQ_BACK, VALID_MCQ_OPTIONS);
+        assertEquals(expectedMcqCard, ParserUtil.parseMcqCard(VALID_MCQ_FRONT, VALID_MCQ_BACK, VALID_MCQ_OPTIONS));
+    }
+
+    @Test
+    public void parseFront_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> ParserUtil.parseFront((Optional<String>) null));
+    }
+
+    @Test
+    public void parseFront_invalidValue_throwsIllegalValueException() {
+        Assert.assertThrows(IllegalValueException.class, () -> ParserUtil.parseFront(Optional.of(WHITESPACE)));
+    }
+
+    @Test
+    public void parseFront_optionalEmpty_returnsOptionalEmpty() throws Exception {
+        assertFalse(ParserUtil.parseFront(Optional.empty()).isPresent());
+    }
+
+    @Test
+    public void parseFront_validValueWithoutWhitespace_returnsFront() throws Exception {
+        assertEquals(Optional.of(VALID_FRONT), ParserUtil.parseFront(Optional.of(VALID_FRONT)));
+    }
+
+    /*
+    @Test
+    public void parseFront_validValueWithWhitespace_returnsTrimmedFront() throws Exception {
+        String frontWithWhitespace = WHITESPACE + VALID_FRONT + WHITESPACE;
+        assertEquals(Optional.of(VALID_FRONT), ParserUtil.parseFront(Optional.of(frontWithWhitespace)));
+    }
+    */
+
+    @Test
+    public void parseBack_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> ParserUtil.parseBack((Optional<String>) null));
+    }
+
+    @Test
+    public void parseBack_invalidValue_throwsIllegalValueException() {
+        Assert.assertThrows(IllegalValueException.class, () -> ParserUtil.parseBack(Optional.of(WHITESPACE)));
+    }
+
+    @Test
+    public void parseBack_optionalEmpty_returnsOptionalEmpty() throws Exception {
+        assertFalse(ParserUtil.parseBack(Optional.empty()).isPresent());
+    }
+
+    @Test
+    public void parseBack_validValueWithoutWhitespace_returnsFront() throws Exception {
+        assertEquals(Optional.of(VALID_BACK), ParserUtil.parseBack(Optional.of(VALID_BACK)));
+    }
+
+    /*
+    @Test
+    public void parseBack_validValueWithWhitespace_returnsTrimmedBack() throws Exception {
+        String frontWithWhitespace = WHITESPACE + VALID_BACK + WHITESPACE;
+        assertEquals(Optional.of(VALID_BACK), ParserUtil.parseBack(Optional.of(backWithWhitespace)));
+    }
+    */
```

Comment:
Is this a temporary removal? If this is intended to be permanent then we can just delete these commented lines entirely

Reply:
Yup just a temporary removal because I haven't got it to work yet

Reply:
roger~